docs/developer/feeds.rst

```diff
@@ -4,122 +4,199 @@
+how long each frame should be in seconds.
+An example can be seen in LS372_agent.py::
+
+    agg_params = {
+        'frame_length': 10*60 #[sec]
+    }
+    self.agent.register_feed('temperatures',
+                             record=True,
+                             agg_params=agg_params,
+                             buffer_time=1)
+
+.. _feed_message_format:
+
+Recorded Feed Message Format
+`````````````````````````````
+Recorded feeds require data to have a specific structure so that the aggregator
+can encode the data into G3 objects.
+Each message published should contain a ``block`` of data, or a group of data
+that is co-sampled. Messages should have this structure::
+
+    message = {
+        'block_name': <Key to identify group of co-sampled data>
+        'timestamp': <ctime of data>
+        'data': {
+            'field_name_1': <datapoint1>,
+            'field_name_2': <datapoint2>
+        }
+    }
+
+or if data is buffered on the agent-side, multiple co-sampled data points can
+be passed at once as::
+
+    message = {
+        'block_name': <Key to identify group of co-sampled data>
+        'timestamps': [ctime1, ctime2, ... ]
+        'data': {
+            'field_name_1': [data1_1, data1_2, ...],
+            'field_name_2': [data2_1, data2_2, ...]
+        }
+    }
+
+Note the pluralized ``timestamps`` key.
+
+Data with consistent ``block_names`` will be written to disk as a single
+``G3TimesampleMap`` object, which stores co-sampled data as a map containing
+multiple G3Vector objects along with a vector of timestamps.
+In the example above, the keys of the ``G3TimesampleMap`` will be
+``field_name_1`` and ``field_name_2``.
+The ``block_name`` is only used internally and will not be written
+to disk.
+
+Each set of data that a feed publishes that is non-cosampled should be
+published to a different ``block_name``.
+For instance, for the L372 agent data coming from separate channels are not
+co-sampled.
+The LS372 temperatures should then look like::
+
+    message = {
+        'block_name': 'channel_01',
+        'timestamp': <ctime>,
+        'data': {
+            'channel_01_T': <channel 1 temperature reading>,
+            'channel_01_V': <channel 1 voltage reading>
+         }
+    }
+
+The LS372 G3Frames will then contain a G3TimesampleMap for each channel,
+containing the temperature and voltage readings along with their timestamps.
+
+Field Name Requirements
+''''''''''''''''''''''''
+Field names must:
+
+- Contain only letters, numbers, and underscores.
+- Begin with a letter or any number of underscores followed by a letter.
+- Be no longer than 255 characters.
+
+Attempting to publish an invalid field name should raise an error by the agent.
+However, if invalid field names somehow make it to the aggregator, the
+aggregator will attempt to correct them before writing to disk.
 
+Subscribing to a Feed
+---------------------
 
+There are a few different ways for your agent to subscribe to an OCS Feed.
```

Comment:
Perhaps mention why an agent might want to subscribe to a feed?  And then point out that non-agents (e.g. clients) can subscribe to feeds, too?  The approach for clients is different, of course, but can be left as a stub (or linked to Clients section, as a stub).